lib/class-wp-theme-json.php

```diff
@@ -910,10 +940,18 @@ function ( $element ) {
 	 * Returns the stylesheet that results of processing
 	 * the theme.json structure this object represents.
 	 *
+	 * @param string $type Type of stylesheet we want accepts 'all', 'block_styles', and 'css_variables'.
 	 * @return string Stylesheet.
 	 */
-	public function get_stylesheet() {
-		return array_reduce( $this->contexts, array( $this, 'to_stylesheet' ), '' );
+	public function get_stylesheet( $type = 'all' ) {
```

Comment:
I'd like us to keep adding tests for the new things. In this case, get_stylesheet( 'block_styles' ) and get_stylesheet( 'css_variables' ). I believe that would have caught the issue with the empty rulesets.